### DISCUSSION TITLE
How to best improve the compendium Navigator (a.ka. the FSS Navigator)

### DISCUSSION THREAD

**Discussion by ahcvankampen** [EDS-Bioinformatics-Laboratory/ENCORE#23] [General]
The navigator is a Python program (https://github.com/EDS-Bioinformatics-Laboratory/FSS-Navigator) that scans the sFSS and builds a project tree and generates a web-page (Navigate.html) containing to explore the project. Currently, the functionality of the navigator is limited. In addition, the presentation of information is not always efficient (e.g., figures, tables), and its behavior depends on the browser. 
A number of issues and possible enhancements are listed in Issues in the Navigator GitHub Repository (https://github.com/EDS-Bioinformatics-Laboratory/FSS-Navigator/issues).

For our Lipidomics ENCORE project a much more comprehensive Navigator has been developed that allows to browse/inspect results from lipidomics analyses. However, this lipidomics navigator is now interwoven with the ENCORE structure and only works with R. Therefore, it is not ideal as a generic tool for ENCORE.

The further development of the Navigator should include but is not limited to: (i) configuration of panel locations (including the possibility to undock to a full window), (ii) Improve the presentation of figures and tables. (iii) improved possibilities to browse results and link results with code and data. (iv)  Ensure proper formatting of text-based information (e.g., markdown, code) while ensuring that all relative hyperlinks work in the context of the Navigator.  This requires a redesign of the navigator since the current setup may not allow all these changes.

**Comment by ahcvankampen** [comment]
Should we consider the use of standard web development frameworks such as Django (www.djangoproject.com) and Vue (Vue.js)?

**Reply by ahcvankampen** [reply]
Are these frameworks not an overkill? Too complex?

**Reply by utkarsh0493** [reply]
While it may seem like too much at the moment, sticking with Django or a similar framework will set the stage for many future improvements in the application.
